tmt/base.py

```diff
@@ -1492,6 +1506,20 @@ def export(self, format_='yaml'):
             raise tmt.utils.GeneralError(
                 f"Invalid plan export format '{format_}'.")
 
+    @property
+    def imports_plan(self) -> bool:
+        return self._import_plan_node is not None
+
+    @property
+    def imported_plan(self) -> Optional['Plan']:
+        if not self.imports_plan:
+            return None
+        if not self._imported_plan:
+            node = fmf.Tree.node(self._import_plan_node)
```

Comment:
The `fmf.Tree.node()` approach can be used for `tmt plan show` (which is using `fmf` cache functionality and stores fetched repositories under `~/.cache/fmf`) but not for real test execution. In order to make the `tmt run` command work with remote plans, the whole git repository has to be fetched into the workdir as suggested [here](https://github.com/teemtee/tmt/pull/1462/commits/d5ddda016a51070a70a473e95520508c459c5796#diff-ed47764f2b7ec3e702c50a15b2e19b3dd731c8c1b7c184dfb7228869bfe09f8aR1061).

There will be some additional work needed to make the test execution working. Please, extend the test to cover the whole plan execution, not only `tmt run discover` which does not reveal this problem of missing files.

Reply:
I added the support to clone the repo manually when the run obj is set. However, I think I did not write a proper test, which verifies this behavior. All remote plans reference plans with tests defined in the plan.fmf; thus, I did not find a problem. Are there quick tests in tmt/tests with a deep directory structure that verifies the correct implementation of this?

After `tmt run discover` there is a test `tmt run -a`. This already runs the whole tmt execution right? So, it is covered right ?

Reply:
Yes, `tmt run` covers that but there were still a couple of adjustments needed to make it work with full test metadata discovery.